src/dataplane-protocol/src/record.rs

```diff
@@ -480,7 +444,11 @@ mod test {
         use crate::record::DefaultRecord;
 
         fn create_batch() -> DefaultBatch {
-            let record: DefaultRecord = vec![0x74, 0x65, 0x73, 0x74].into();
+            let value = vec![0x74, 0x65, 0x73, 0x74];
+            let record = DefaultRecord {
+                value: DefaultAsyncBuffer::new(value),
```

Comment:
Should not use `DefaultAsyncBuffer`.   It should be something like:
`DefaultRecord::new(key, value)`